### PR TITLE
1747 show shouts

### LINKED ISSUE #1747: 🚀 [Feature] Article's Shout displayed in Profile or not
> [datenbrei](https://github.com/datenbrei) **Authored by [datenbrei](https://github.com/datenbrei)**
_<time datetime="2019-09-27T11:32:05Z" title="Friday, September 27th 2019, 1:32:05 pm +02:00">Sep 27, 2019</time>_
_Closed <time datetime="2019-11-18T16:00:44Z" title="Monday, November 18th 2019, 5:00:44 pm +01:00">Nov 18, 2019</time>_
---

## :rocket: Feature
If somebody shouts an article it should be possible to decide, if this shout should be also displayed in the profile or not.

### Additional context
This is a privacy option. Somebody could just shout non-political articles for example so that nobody could guess his political direction from it.

### PR DESCRIPTION
**Pull request by Tirokk**
> [Kachulio1](https://github.com/Kachulio1) **Authored by [Kachulio1](https://github.com/Kachulio1)**
_<time datetime="2019-10-31T18:27:11Z" title="Thursday, October 31st 2019, 7:27:11 pm +01:00">Oct 31, 2019</time>_
_Merged <time datetime="2019-11-18T16:00:44Z" title="Monday, November 18th 2019, 5:00:44 pm +01:00">Nov 18, 2019</time>_
---

## 🍰 Pull request
Allows the user to set shouts either public or private

### Issues
- fixes #1747 

### Todo
- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [alina-beck](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-04T11:53:36Z" title="Monday, November 4th 2019, 12:53:36 pm +01:00">Nov 4, 2019</time>_

---


Hey @Kachulio1! Matt said you want some frontend support on this? Do you have any implementation details in mind already?

I think we would need to:
- [x] add this to the settings page
- [x] toggle the `shouts` tab on the user profile depending on the setting
- [x] display a list of shouted posts when the `shouts` tab is selected

Did I miss anything?

**Comment by Tirokk** [comment]
> [Kachulio1](https://github.com/Kachulio1) **Authored by [Kachulio1](https://github.com/Kachulio1)**
_<time datetime="2019-11-13T18:35:35Z" title="Wednesday, November 13th 2019, 7:35:35 pm +01:00">Nov 13, 2019</time>_

---


@alina-beck sorry have been tied up with personal things. sorry for the late response.
should we sync on Monday about things I should work on the backend?

**Comment by Tirokk** [comment]
> [alina-beck](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-14T06:37:09Z" title="Thursday, November 14th 2019, 7:37:09 am +01:00">Nov 14, 2019</time>_

---


Hey @Kachulio1, no problem! I think it's done, actually. We already did fetch `shouted posts` because we had the feature before, so I just needed to toggle them on the profile. And the `mutation` you set up worked fine. :+1:

File: backend/src/schema/resolvers/users.js [hunk 1]
```diff
@@ -177,6 +177,7 @@ export default {
         'termsAndConditionsAgreedVersion',
         'termsAndConditionsAgreedAt',
         'allowEmbedIframes',
+        'showShoutsPublicly',
```
Comment (Tirokk) [inline backend/src/schema/resolvers/users.js:180]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-14T08:17:53Z" title="Thursday, November 14th 2019, 9:17:53 am +01:00">Nov 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -177,6 +177,7 @@ export default {
         'termsAndConditionsAgreedVersion',
         'termsAndConditionsAgreedAt',
         'allowEmbedIframes',
+        'showShoutsPublicly',
```


Very important that we not forget to add this! Great job catching this issue before it bites us @Kachulio1

**Review by Mogge — APPROVED** [review]
got my approval!! good job!
One thing @Tirokk brought up after our deploy process yesterday introducing another bug,
He said that we could get in the habit of adding tests that set the property we add to undefined and then make sure that it gets mapped to null since undefined breaks the site, and null does not.
I like this idea! do you think you could add such a test @Kachulio1 ??

**Review by Mogge — APPROVED** [review]
we love you

We :heart:  you @Kachulio1 :+1: thank you so much
